tests/testthat/test-dataset.R

```diff
@@ -0,0 +1,105 @@
+library(tidyverse)
+library(testthat)
+library(lubridate)
+
+
+cvpia_watershed <- DSMflow::watershed_ordering$watershed
+
+#Tests in this script test:
+
+#test dimensionality, test for names (watershed rows), test proportion value <= 1,
+#test celisus and f are c and f for its range and type
+#update documentation as test develops
+
+#Temperature structure ---------------------------------------------------------
+test_that("Temperature dimensions", {
+  expect_equal(dim(degree_days), c(31, 12, 22))
+  expect_equal(dim(delta_temperature), c(12, 21, 2))
+  expect_equal(dim(egg_temperature_effect), c(31, 4))
+  expect_equal(dim(migratory_temperature_proportion_over_20), c(31, 12))
+  expect_equal(dim(prisoners_point_temperature), c(12, 21))
+  expect_equal(dim(stream_temperature), c(31, 12, 21))
+  expect_equal(dim(vernalis_temperature), c(12, 21))
+  })
+
+#Temperature watershed labels --------------------------------------------------
+test_that("Temperature watershed labels are the same as cvpia watershed", {
+  expect_equal(row.names(degree_days), cvpia_watershed)
+  expect_equal(unlist(egg_temperature_effect['watershed'], use.names= FALSE), cvpia_watershed)
+  expect_equal(row.names(migratory_temperature_proportion_over_20), cvpia_watershed)
+  expect_equal(row.names(stream_temperature), cvpia_watershed)
+})
+
+# DSMtemperature::migratory_temperature_proportion_over_20 ---------------------
+migratory_temperature_proportion_over_20_df <- migratory_temperature_proportion_over_20 %>%
+  as.data.frame() %>%
+  rownames_to_column(var = 'location') %>%
+  pivot_longer(!location, names_to = "month", values_to = "migratory_temperature_proportion_over_20")
+
+test_that("proportion less or equal to 1", {
+  expect_true(all(migratory_temperature_proportion_over_20_df['migratory_temperature_proportion_over_20'] >= 0))
+  expect_true(all(migratory_temperature_proportion_over_20_df['migratory_temperature_proportion_over_20'] <= 1))
+})
+
+#DSMtemperature::delta_temperature ---------------------------------------------
+delta_temperature_df <- map_df(c("North Delta", "South Delta"), function(i){
+  delta_temperature[, , i] %>%
+    as.data.frame() %>%
+    mutate(location = i) %>%
+    rownames_to_column(var = 'month') %>%
+    pivot_longer(cols = '1980':'2000',
+                 names_to ="year",
+                 values_to = 'delta_temperature')}) %>%
+  pivot_wider(id_cols = c('month','year'),
+              names_from = location,
+              values_from = delta_temperature) %>%
+    rename(n_dlt_delta_temperature = 'North Delta',
+           s_dlt_delta_temperature = 'South Delta')
+
+max_river_celsius <- 50
```

Comment:
Here you test that all fall below a max river Celsius. Would it be good to also test that they are all greater than a min?

Reply:
min added!